download/download.go

```diff
@@ -26,6 +26,13 @@ const (
 	minRetryDelay = time.Millisecond * 100
 )
 
+// Interface can be implemented to download via arbitrary transports/formats.
+type Interface interface {
```

Comment:
🤔 just thinking out loud a little bit, `download.Interface` is a little.. i dunno.. maybe too close to just `interface{}` or something. Any thoughts on calling it something more like `download.Client` or `download.Transport` or something?

The first time I read through the PR I was just like "ah yea, its the download interface" but now I'm having some second thoughts on it.